doc/user/compatibility.md

```diff
@@ -104,6 +104,14 @@ Using `set_trace_func` will temporarily lower the performance of your program.
 As with `ObjectSpace`, we would recommend that you do not use this in the inner
 loop of your production application.
 
+#### Backtraces
+
+Throwing exceptions, and other operations which need to create a backtrace, are
+slower than on MRI. This is because we have to undo optimisations that we have
+applied to run your Ruby code fast in order to recreate the backtrace entries.
+We wouldn't recommend using exceptions for control flow on any implementation of
+Ruby anyway.
```

Comment:
Maybe we could mention the `rescue` form not using the backtrace, or `throw`/`catch` but it's probably too much details for this document.